{
	"schema_version": "1.6.0",
	"id": "UBUNTU-CVE-2020-36242",
	"modified": "2021-02-07T20:15:00Z",
	"published": "2021-02-07T20:15:00Z",
	"related": [
		"CVE-2020-36242"
	],
	"details": "In the cryptography package before 3.3.2 for Python, certain sequences of update calls to symmetrically encrypt multi-GB values could result in an integer overflow and buffer overflow, as demonstrated by the Fernet class.",
	"severity": [
		{
			"type": "CVSS_V3",
			"score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:H"
		}
	],
	"affected": [
		{
			"package": {
				"ecosystem": "Ubuntu:16.04:LTS",
				"name": "python-cryptography",
				"purl": "pkg:deb/ubuntu/python-cryptography?arch=src?distro=xenial"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1.2.3-1ubuntu0.3"
						}
					]
				}
			],
			"versions": [
				"1.0.1-1ubuntu1",
				"1.0.2-1",
				"1.1-1",
				"1.1.1-1",
				"1.1.1-1ubuntu1",
				"1.1.1-1ubuntu2",
				"1.1.1-1ubuntu3",
				"1.2.2-2",
				"1.2.3-1",
				"1.2.3-1ubuntu0.1",
				"1.2.3-1ubuntu0.2"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-cryptography",
						"binary_version": "1.2.3-1ubuntu0.3"
					},
					{
						"binary_name": "python-cryptography-doc",
						"binary_version": "1.2.3-1ubuntu0.3"
					},
					{
						"binary_name": "python3-cryptography",
						"binary_version": "1.2.3-1ubuntu0.3"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-36242.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:18.04:LTS",
				"name": "python-cryptography",
				"purl": "pkg:deb/ubuntu/python-cryptography?arch=src?distro=bionic"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "2.1.4-1ubuntu1.4"
						}
					]
				}
			],
			"versions": [
				"1.9-1",
				"2.1.3-3",
				"2.1.4-1",
				"2.1.4-1build1",
				"2.1.4-1build2",
				"2.1.4-1ubuntu1",
				"2.1.4-1ubuntu1.1",
				"2.1.4-1ubuntu1.2",
				"2.1.4-1ubuntu1.3"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-cryptography",
						"binary_version": "2.1.4-1ubuntu1.4"
					},
					{
						"binary_name": "python-cryptography-dbgsym",
						"binary_version": "2.1.4-1ubuntu1.4"
					},
					{
						"binary_name": "python-cryptography-doc",
						"binary_version": "2.1.4-1ubuntu1.4"
					},
					{
						"binary_name": "python3-cryptography",
						"binary_version": "2.1.4-1ubuntu1.4"
					},
					{
						"binary_name": "python3-cryptography-dbgsym",
						"binary_version": "2.1.4-1ubuntu1.4"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-36242.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:20.04:LTS",
				"name": "python-cryptography",
				"purl": "pkg:deb/ubuntu/python-cryptography?arch=src?distro=focal"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "2.8-3ubuntu0.1"
						}
					]
				}
			],
			"versions": [
				"2.6.1-3.1",
				"2.6.1-4",
				"2.6.1-4ubuntu1",
				"2.8-1ubuntu2",
				"2.8-2",
				"2.8-3"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-cryptography",
						"binary_version": "2.8-3ubuntu0.1"
					},
					{
						"binary_name": "python-cryptography-dbgsym",
						"binary_version": "2.8-3ubuntu0.1"
					},
					{
						"binary_name": "python-cryptography-doc",
						"binary_version": "2.8-3ubuntu0.1"
					},
					{
						"binary_name": "python3-cryptography",
						"binary_version": "2.8-3ubuntu0.1"
					},
					{
						"binary_name": "python3-cryptography-dbgsym",
						"binary_version": "2.8-3ubuntu0.1"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-36242.json"
			}
		}
	],
	"references": [
		{
			"type": "REPORT",
			"url": "https://ubuntu.com/security/CVE-2020-36242"
		},
		{
			"type": "REPORT",
			"url": "https://github.com/pyca/cryptography/blob/master/CHANGELOG.rst"
		},
		{
			"type": "REPORT",
			"url": "https://github.com/pyca/cryptography/compare/3.3.1...3.3.2"
		},
		{
			"type": "REPORT",
			"url": "https://www.cve.org/CVERecord?id=CVE-2020-36242"
		}
	]
}
